infra/terraform/vpc_hub/README.md

```diff
@@ -0,0 +1,73 @@
+# VPC Hub
+
+This Terraform Module creates a VPC Hub for VPC peering connection to VPC Spokes. The goal of this VPC is to centralize VPC PrivateLink connections to AWS services so we don't have to create VPC PrivateLink connections in each VPC Spoke for all services we use. This will allow us to only pay for the VPC Interface Endpoints in the VPC Hub once.
+
+A VPC Spoke is simply just another VPC which wants to access AWS services without going through the public internet.
+
+```mermaid
+graph LR
+    S[AWS Service]
+    H[VPC Hub]
+    V1[VPC Spoke X]
+    V2[VPC Spoke Y]
+
+    H -- PrivateLink <--> S
+    V1 -- VPC Peering Connection <--> H
+    V2 <--> H
+
+```
+
+## Considerations for using the VPC Hub
+
+When a VPC Spoke is to be connected to the Hub, it's essential that theres a non-ambiguous network route to the Spoke. Essentially this means our infrastructure should have unique CIDR blocks for each environment in every project. If this is not the case, the VPC Hub will not know which Spoke to route traffic back to. I suggest reading this [AWS documentation site](https://docs.aws.amazon.com/vpc/latest/peering/peering-configurations-partial-access.html#peering-incorrect-response-routing) to understand why this is the case.
+
+If we were to use this VPC Hub, we have to make an active decision to use unique CIDR blocks for each environment in every project in AWS. A common CIDR delegation scheme such as the following would be beneficial to stick to:
+
+```text
+10.0.0.0/14 ~ [Project Root] ~ e.g. 10.0.0.0 - 10.3.255.255
+    This is a logical part of the network and never actually allocated in AWS.
```

Comment:
Didn't quite follow this comment – does it mean that this will be the implicit CIDR block of all environments in a project, but we never actually write `10.x.x.x/14` anywhere in a Terraform file/AWS config? wouldn't we need `10.0.0.0/14` as the CIRD block for a Project that contains 4 environment, where each environment is taking up `/16`? I may have misunderstood all this

Reply:
Yes, you got it right. We can't delegate that /14 because AWS doesn't allow more than a `/16` for a VPC. So we'd have to manage our numbering separately. The /14 doesn't need to be allocated anywhere because each VPC would just take on `10.0../16`, `10.1../16`, `10.2../16`, `10.3../16` for each environment.

Ideally we would be able to actually allocate a `/14`, but at this point it is more complicated than keeping track of the numbers per project. I think ideally we'd be able to have some shared Terraform module that has all project names so we can at least statically define the delegation. Maybe it should be an array of project names and calculate the project `/14` from the index in the list